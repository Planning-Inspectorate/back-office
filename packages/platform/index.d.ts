import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
