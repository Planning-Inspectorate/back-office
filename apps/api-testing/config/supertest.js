// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
