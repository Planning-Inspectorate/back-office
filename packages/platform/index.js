export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
