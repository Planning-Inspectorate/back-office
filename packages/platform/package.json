{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
