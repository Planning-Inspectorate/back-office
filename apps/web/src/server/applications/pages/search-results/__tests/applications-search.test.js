import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const successFullResponse = {
	page: 1,
	pageSize: 1,
	pageCount: 1,
	itemCount: 1,
	items: [
		{
			id: 76,
			reference: null,
			modifiedDate: '1662373959',
			title: 'Title',
			status: 'Draft',
			publishedDate: null
		}
	]
};

const successEmptyResponse = {
	page: 1,
	pageSize: 1,
	pageCount: 1,
	itemCount: 0,
	items: []
};

describe('applications search', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	beforeEach(async () => {
		await request.get('/applications-service/case-officer');
	});

	describe('POST /applications-service/search-results', () => {
		const baseUrl = '/applications-service/search-results';

		it('should render the page with the results if params are correct', async () => {
			nock('http://test/').post('/applications/search').reply(200, successFullResponse);

			const response = await request.post(baseUrl).send({ query: 'query' });
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain(
				'<a class="govuk-body" href="/applications-service/create-new-case/76/check-your-answers"> Title</a>'
			);
		});

		it('should render a message if no results are found', async () => {
			nock('http://test/').post('/applications/search').reply(200, successEmptyResponse);

			const response = await request.post(baseUrl).send({ query: 'query' });
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('There are no matching results');
		});

		it('should render a message if the query is empty', async () => {
			const response = await request.post(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Enter a search term');
		});
	});
});
