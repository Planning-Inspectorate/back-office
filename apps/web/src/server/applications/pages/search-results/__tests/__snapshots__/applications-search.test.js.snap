// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications search POST /applications-service/search-results-results should render a message if no results are found 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-full\\">
            <h1 class=\\"govuk-heading-xl\\">Search results</h1>
        </div>
    </div>
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-one-half\\">
            <form method=\\"post\\" action=\\"search-results\\" novalidate class=\\"pins-dashboard-box--form pins-dashboard-form\\">
                <div class=\\"govuk-form-group\\">
                    <input class=\\"govuk-input pins-dashboard-form--input\\" id=\\"searchApplications\\"
                    name=\\"query\\" type=\\"text\\" value=\\"query\\" inputmode=\\"search\\">
                </div>
                <button class=\\"govuk-button pins-dashboard-form--btn\\" data-module=\\"govuk-button\\">Search</button>
            </form>
        </div>
    </div>
    <div>
        <p class=\\"govuk-heading-s\\">0 results</p>
        <hr class=\\"govuk-section-break govuk-section-break--l govuk-section-break--visible\\">
        <p class=\\"govuk-heading-s\\">There are no matching results</p>
        <p class=\\"govuk-body\\">Improve your search by:</p>
        <ul class=\\"govuk-list govuk-list--bullet\\">
            <li>double-checking your spelling</li>
            <li>make sure you have the case reference number right</li>
        </ul>
    </div>
</main>"
`;

exports[`applications search POST /applications-service/search-results-results should render a message if the query is empty 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-full\\">
            <h1 class=\\"govuk-heading-xl\\">Search results</h1>
        </div>
    </div>
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-one-half\\">
            <form method=\\"post\\" action=\\"search-results\\" novalidate class=\\"pins-dashboard-box--form pins-dashboard-form\\">
                <div class=\\"govuk-form-group govuk-form-group--error\\">
                    <p id=\\"searchApplications-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span> Enter a search term</p>
                    <input                     class=\\"govuk-input pins-dashboard-form--input govuk-input--error\\" id=\\"searchApplications\\"
                    name=\\"query\\" type=\\"text\\" aria-describedby=\\"searchApplications-error\\" inputmode=\\"search\\">
                </div>
                <button class=\\"govuk-button pins-dashboard-form--btn\\" data-module=\\"govuk-button\\">Search</button>
            </form>
        </div>
    </div>
    <div>
        <p class=\\"govuk-heading-s\\">0 results</p>
    </div>
</main>"
`;

exports[`applications search POST /applications-service/search-results-results should render the page with the results if params are correct 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-full\\">
            <h1 class=\\"govuk-heading-xl\\">Search results</h1>
        </div>
    </div>
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-one-half\\">
            <form method=\\"post\\" action=\\"search-results\\" novalidate class=\\"pins-dashboard-box--form pins-dashboard-form\\">
                <div class=\\"govuk-form-group\\">
                    <input class=\\"govuk-input pins-dashboard-form--input\\" id=\\"searchApplications\\"
                    name=\\"query\\" type=\\"text\\" value=\\"query\\" inputmode=\\"search\\">
                </div>
                <button class=\\"govuk-button pins-dashboard-form--btn\\" data-module=\\"govuk-button\\">Search</button>
            </form>
        </div>
    </div>
    <div>
        <p class=\\"govuk-heading-s\\">1 results</p>
        <table class=\\"govuk-table pins-table--fixed pins-table--bordered\\">
            <thead class=\\"govuk-table__head\\">
                <tr class=\\"govuk-table__row\\">
                    <th scope=\\"col\\" class=\\"govuk-table__header govuk-!-width-one-half\\">Name</th>
                    <th scope=\\"col\\" class=\\"govuk-table__header text-align--right\\">Stage</th>
                </tr>
            </thead>
            <tbody class=\\"govuk-table__body\\">
                <tr class=\\"govuk-table__row\\">
                    <td class=\\"govuk-table__cell\\"><a class=\\"govuk-body\\" href=\\"/applications-service/create-new-case/76/check-your-answers/\\"> Title</a>
                    </td>
                    <td class=\\"govuk-table__cell govuk-!-width-one-quarter text-align--right\\"><strong class=\\"govuk-tag govuk-tag--\\">Draft</strong>
                    </td>
                </tr>
            </tbody>
        </table>
    </div>
</main>"
`;
